server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectWriteAheadLogs.java

```diff
@@ -80,10 +80,10 @@
 
 public class GarbageCollectWriteAheadLogs {
   private static final Logger log = LoggerFactory.getLogger(GarbageCollectWriteAheadLogs.class);
+  private static final Map<HostAndPort,Long> firstSeenDead = new HashMap<HostAndPort,Long>();
 
   private final AccumuloServerContext context;
   private final VolumeManager fs;
-  private final Map<HostAndPort,Long> firstSeenDead = new HashMap<HostAndPort,Long>();
```

Comment:
I tried really hard to make an integration test showed the initial error reported in ACCUMULO-4157, but it was too hard to control the moving parts make a WAL be deleted before it should have been.  My work is [here](https://github.com/mjwall/accumulo/blob/62b0e879cbc00773d1fbdbada9d696d42ecd6531/test/src/test/java/org/apache/accumulo/test/functional/GarbageCollectWriteAheadLogsIT.java) if you want to look.  It is ugly and didn't work, but maybe will give you starting point.  Or maybe it is just an example of what not to do.